lib/codeowners/checker/group.rb

```diff
@@ -67,7 +67,9 @@ def owners
 
       def subgroups_owned_by(owner)
         @list.flat_map do |item|
-          return [] unless item.is_a?(Group)
+          unless item.is_a?(Group)
+            item.parent.parent ? (return []) : next
```

Comment:
hmm, why not just `next`?

Reply:
When we have nested groups, instead of the name of the group it would return also all the subgroups. Do you think it's desirable?

Reply:
I guess yes because we're probably going to push something in the sub-sub-group level.

I'm still struggling to understand the algorithm at all. because the `return` part got me very confused.

what about using a guard clause and make it return early?

```ruby
def subgroups_owned_by(owner)
   return [] if @list.flatten.grep(Group).any?{|item|item.parent.parent }
   ...
end
```

Reply:
Ok, we can return all sub-subgroups and use `next`, we don't need the `parent.parent`.